### PR TITLE
Hotfix/fixes

### PR DESCRIPTION
**Pull request by marchermans**
Some critical hotfixes:
- Split requests are now handled better
- Placementhandlers will now use a partial amount in the inventory first
- Warehouse only delivers requested amount.

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java [hunk 1]
```diff
@@ -38,27 +38,27 @@
...
+     * Method to create a new {@link CombinedItemHandler}.
      *
-     * @param defaultName The default name of this {@link AbstractCombinedItemHandler}.
+     * @param defaultName The default name of this {@link CombinedItemHandler}.
      * @param handlers    The combining {@link IItemHandlerModifiable}.
      */
-    protected AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final IItemHandlerModifiable... handlers)
+    public CombinedItemHandler(@NotNull final String defaultName, @NotNull final IItemHandlerModifiable... handlers)
     {
         this.handlers = handlers;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java:48]: MINOR Store a copy of "handlers". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)

File: src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java [hunk 2]
```diff
@@ -262,6 +262,21 @@ public ItemStack extractItem(final int slot, final int amount, final boolean sim
...
 
+    @Override
+    public int getSlotLimit(final int slot)
+    {
+        int slotIndex = slot;
+        for (final IItemHandlerModifiable modifiable : handlers)
+        {
+            if (slotIndex >= modifiable.getSlots())
+                slotIndex-=modifiable.getSlots();
+            else
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java:273]: CRITICAL Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)

File: src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java [hunk 3]
```diff
@@ -262,6 +262,21 @@ public ItemStack extractItem(final int slot, final int amount, final boolean sim
...
+                return modifiable.getSlotLimit(slotIndex);
+        }
+
+        return 0;
+    }
+
     protected IItemHandlerModifiable[] getHandlers()
     {
         return handlers;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java:282]: MINOR Return a copy of "handlers". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)

File: src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java [hunk 4]
```diff
@@ -262,6 +262,21 @@ public ItemStack extractItem(final int slot, final int amount, final boolean sim
         return null;
     }
 
+    @Override
+    public int getSlotLimit(final int slot)
+    {
+        int slotIndex = slot;
+        for (final IItemHandlerModifiable modifiable : handlers)
+        {
+            if (slotIndex >= modifiable.getSlots())
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java:271]: CRITICAL Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)

File: src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java [hunk 5]
```diff
@@ -38,27 +38,27 @@
...
+     * @param defaultName The name of this {@link CombinedItemHandler}.
      * @param customName  The preset custom name of this {@link
-     *                    AbstractCombinedItemHandler}.
+     *                    CombinedItemHandler}.
      * @param handlers    The combinging {@link IItemHandlerModifiable}.
      */
-    protected AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final String customName, @NotNull final IItemHandlerModifiable... handlers)
+    public CombinedItemHandler(@NotNull final String defaultName, @NotNull final String customName, @NotNull final IItemHandlerModifiable... handlers)
     {
         this.handlers = handlers;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/inventory/api/CombinedItemHandler.java:63]: MINOR Store a copy of "handlers". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 6]
```diff
@@ -1694,15 +1701,67 @@ public void onRequestCancelled(@NotNull final IToken token)
...
+
     //------------------------- !END! RequestSystem handling for minecolonies buildings -------------------------//
+
+    //------------------------- !Start! Capabilities handling for minecolonies buildings -------------------------//
+
+    @Override
+    public boolean hasCapability(
+      @Nonnull final Capability<?> capability, @Nullable final EnumFacing facing)
+    {
+        if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY && facing == null)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:1728]: MINOR Replace this if-then-else statement by a single return statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 7]
```diff
@@ -71,7 +78,7 @@
  * to their views and blocks.
  */
 @SuppressWarnings("squid:S2390")
-public abstract class AbstractBuilding implements IRequestResolverProvider, IRequester
+public abstract class AbstractBuilding implements IRequestResolverProvider, IRequester, ICapabilityProvider
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:81]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 61 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "AbstractBuilding" has 92 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java [hunk 8]
```diff
@@ -80,6 +85,30 @@ public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequ
...
+            {
+                AbstractBuilding building = (AbstractBuilding) ((BuildingBasedRequester) request.getRequester()).getBuilding().get();
+                Optional<CitizenData> citizenDataOptional = building.getCitizenForRequest(request.getToken());
+
+                List<ItemStack> resolvablestacks = request.getDisplayStacks();
+                if (!resolvablestacks.isEmpty() && citizenDataOptional.isPresent())
+                {
+                    ItemStack resolveStack = resolvablestacks.get(0);
+                    resolveStack.setCount(Math.min(((IDeliverable) request.getRequest()).getCount(), resolveStack.getMaxStackSize()));
+                    ItemStack remainingItemStack = InventoryUtils.addItemStackToItemHandlerWithResult(new InvWrapper(citizenDataOptional.get().getCitizenEntity().getInventoryCitizen()), resolveStack);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java]: MAJOR Split this 200 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java [hunk 9]
```diff
@@ -80,6 +85,30 @@ public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequ
         final IColony colony = manager.getColony();
         if (colony instanceof Colony)
         {
+            if (Configurations.requestSystem.creativeResolve &&
+                  request.getRequest() instanceof IDeliverable &&
+                  request.getRequester() instanceof BuildingBasedRequester &&
+                  ((BuildingBasedRequester) request.getRequester()).getBuilding().isPresent() &&
+                  ((BuildingBasedRequester) request.getRequester()).getBuilding().get() instanceof AbstractBuilding)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 10]
```diff
@@ -495,18 +495,20 @@ private AIState lookForRequests()
...
+                if(InventoryUtils.getItemCountInItemHandler(
+                        new InvWrapper(worker.getInventoryCitizen()), deliveredItemStack::isItemEqualIgnoreDurability) >= deliveredItemStack.getCount())
                 {
-                    getOwnBuilding().markRequestAsAccepted(worker.getCitizenData(), firstDeliverableRequest.getToken());
                     return NEEDS_ITEM;
                 }
 
                 //Takes one Stack from the hut if existent
-                if (isInHut(deliveredItemStack)  && !walkToBuilding())
+                if (! walkToBuilding() && InventoryUtils.getItemCountInProvider(getOwnBuilding(), deliveredItemStack::isItemEqualIgnoreDurability) >= deliveredItemStack.getCount() &&
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: MAJOR Split this 182 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 11]
```diff
@@ -495,18 +495,20 @@ private AIState lookForRequests()
...
+                  InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(getOwnBuilding(), deliveredItemStack::isItemEqualIgnoreDurability, deliveredItemStack.getCount(), new InvWrapper(worker.getInventoryCitizen())))
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: MAJOR Split this 241 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 97 issues
* ![CRITICAL][CRITICAL] 10 critical
* ![MAJOR][MAJOR] 26 major
* ![MINOR][MINOR] 59 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1438](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1438): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1467](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1467): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1478](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1478): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1581](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1581): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1695](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1695): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [StandardPlayerRequestResolver.java#L186](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L186): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [WarehouseRequestResolver.java#L140](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java#L140): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [WarehouseRequestResolver.java#L146](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java#L146): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#): This file has 1.122 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [StandardPlayerRequestResolver.java#L78](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L78): Return an empty collection instead of null. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 97 issues
* ![CRITICAL][CRITICAL] 10 critical
* ![MAJOR][MAJOR] 26 major
* ![MINOR][MINOR] 59 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1438](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1438): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1467](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1467): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1478](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1478): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1581](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1581): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1695](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1695): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [StandardPlayerRequestResolver.java#L186](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L186): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [WarehouseRequestResolver.java#L140](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java#L140): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [WarehouseRequestResolver.java#L146](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java#L146): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#): This file has 1,122 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [StandardPlayerRequestResolver.java#L78](https://github.com/Minecolonies/minecolonies/blob/b4a0957938018e21ff0c21916c50a7f1f19a309e/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L78): Return an empty collection instead of null. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java (hunk 10 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: OKay this is the only thing which bothers me, can we make it that the worker always walks to the building when he needs something and he waits there for the item?

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1864/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java [hunk 12]
```diff
@@ -80,6 +85,32 @@ public void resolve(@NotNull final IRequestManager manager, @NotNull final IRequ
...
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java:92]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3655)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 94 issues
* ![CRITICAL][CRITICAL] 10 critical
* ![MAJOR][MAJOR] 23 major
* ![MINOR][MINOR] 59 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1438](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1438): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1467](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1467): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1478](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1478): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1581](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1581): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1695](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1695): Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)
1. ![CRITICAL][CRITICAL] [StandardPlayerRequestResolver.java#L188](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L188): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [WarehouseRequestResolver.java#L140](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java#L140): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [WarehouseRequestResolver.java#L146](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java#L146): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#): This file has 1.122 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [StandardPlayerRequestResolver.java#L78](https://github.com/Minecolonies/minecolonies/blob/64c6cb46f20cc4c2a56bcb65c2cf8795f2132181/src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/StandardPlayerRequestResolver.java#L78): Return an empty collection instead of null. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1168)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
Will merge this, we need this tested, and I'll be happy to clean up all those issues.

**Pull request merged**